### ISSUE TITLE
Implement MatheCoachFeedbackGenerator for Human-Readable Tips

### ISSUE THREAD

**Issue by yannicktuerk** [yannicktuerk/F1-Lap-Bot#46]
## Summary
Generate human-readable coaching feedback from LapComparison results. Translates technical deltas into actionable driving advice with track landmarks and physics explanations.

## Technical Details
Create `MatheCoachFeedbackGenerator` in `src/domain/services/mathe_coach_feedback.py`:

### Feedback Template:
```
📊 Lap Analysis: Track Name

⏱️ Total time vs ideal: +2.35s

🎯 Top Improvement Areas:

1️⃣ Turn 3 Entry (0.42s loss)
   • Braking 15m too early
   • Entry speed: 142 km/h (ideal: 156 km/h)
   • 💡 Tip: Trust the grip! You can brake 15m later here.

2️⃣ Sector 2 Straight (0.31s loss)
   • Throttle application 0.3s late
   • Exit speed: 198 km/h (ideal: 215 km/h)
   • 💡 Tip: Get on throttle earlier. The car can handle it.

3️⃣ Turn 6 Apex (0.28s loss)
   • Apex speed: 127 km/h (ideal: 138 km/h)
   • 💡 Tip: Carry more speed through the apex. Physics allows 138 km/h.

📈 Focus on corner exits for maximum gains!
```

### Implementation:
```python
class MatheCoachFeedbackGenerator:
    def generate_feedback(
        self,
        comparison: List[ComparisonSegment],
        track_profile: TrackProfile,
        track_name: str,
        top_n: int = 5
    ) -> str:
        """
        Generate Markdown-formatted coaching feedback.
        """
        total_loss = sum(seg.time_loss for seg in comparison)
        
        lines = [
            f"📊 **Lap Analysis: {track_name}**",
            "",
            f"⏱️ Total time vs ideal: **+{total_loss:.2f}s**",
            "",
            "🎯 **Top Improvement Areas:**",
            ""
        ]
        
        for i, seg in enumerate(comparison[:top_n], 1):
            lines.append(f"{i}️⃣ **{seg.segment_id}** ({seg.time_loss:.2f}s loss)")
            
            # Add specific feedback based on error type
            if seg.error_type == "early_braking":
                lines.append(f"   • Braking too early")
                lines.append(f"   • Entry speed: {seg.speed_delta_entry:+.0f} km/h vs ideal")
                lines.append(f"   • 💡 **Tip:** Brake later! Physics allows more grip here.")
            
            elif seg.error_type == "slow_corner":
                lines.append(f"   • Apex speed: {seg.speed_delta_apex:+.0f} km/h vs ideal")
                lines.append(f"   • 💡 **Tip:** Carry more mid-corner speed. The grip circle allows it.")
            
            elif seg.error_type == "late_throttle":
                lines.append(f"   • Throttle application too late")
                lines.append(f"   • Exit speed: {seg.speed_delta_exit:+.0f} km/h vs ideal")
                lines.append(f"   • 💡 **Tip:** Get on throttle earlier on exit.")
            
            lines.append("")
        
        lines.append("📈 **Overall advice:** Focus on corner exits for maximum time gains!")
        
        return "\n".join(lines)
```

## Checklist
- [ ] Implement error item formatting with track context
- [ ] Map distance to track landmarks (corners, sectors)
- [ ] Generate physics-based explanations for each error
- [ ] Create summary section with total time loss
- [ ] Format output as Markdown for Discord
- [ ] Write unit tests with sample ComparisonSegment data
- [ ] Test readability and actionability of feedback
- [ ] Support multiple languages (English/German)

## Acceptance Criteria
- MatheCoachFeedbackGenerator produces clear, actionable messages
- References specific track locations
- Includes physics principles (grip, braking distance)
- Markdown formatting works in Discord

## References
- Spec: `warp_prompt_f1_lap_bot.md` Section 5.5
